app/controllers/user_controller.rb

```diff
@@ -0,0 +1,35 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: %i[show edit update destroy]
+  before_action :authenticate_user!
+
+  # GET /users/new
+  def new
+    @user = User.new
+  end
+
+  # POST /users or /users.json
+  def create
+    @user = User.new(user_params)
+
+    respond_to do |format|
+      if @user.save
+        format.html { redirect_to user_url(@user), notice: 'User was successfully created.' }
+        format.json { render :show, status: :created, location: @user }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @user.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  private
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_user
+    @user = User.find(params[:id])
```

Comment:
- [ ] [OPTIONAL] For this project you have used the correct coloring scheme and typography but the decision doesn't 100% meet with the design provided to you. Such as a complete page, add new categories page, etc. I recommend you follow the design as much as possible.